RevenueCat/Scripts/Purchases.cs

```diff
@@ -209,28 +289,119 @@ public void GetProducts(string[] products, GetProductsFunc callback, string type
+    /// <param name="oldSku"> Android only. Optional. The oldSku to upgrade from. </param>
+    /// <param name="prorationMode"> Android only. Optional. The <see cref="ProrationMode"/> to use when upgrading the given oldSku. </param>
+    ///
     public void PurchasePackage(Package package, MakePurchaseFunc callback, string oldSku = null,
         ProrationMode prorationMode = ProrationMode.UnknownSubscriptionUpgradeDowngradePolicy)
     {
         MakePurchaseCallback = callback;
         _wrapper.PurchasePackage(package, oldSku, prorationMode);
     }
 
+    ///
+    /// <summary>
+    /// iOS only. Initiates a purchase of a <see cref="Package"/>.
+    /// You can get a <c>PromotionalOffer</c> by calling <see cref="GetPromotionalOffer"/>.
+    /// </summary>
+    ///
+    /// <remarks>
+    /// Call this method when a user has decided to purchase a product.
+    /// Only call this in direct response to user input.
+    /// </remarks>
+    ///
+    /// From here the SDK will handle the purchase with <c>StoreKit</c> and call the <c>PurchaseCompletedBlock</c>.
+    ///
+    /// <remarks>
+    /// Note: You do not need to finish the transaction yourself in the completion callback, RevenueCat will
+    /// handle this for you.
+    /// </remarks>
+    ///
+    /// <param name="package"> The <see cref="Package"/> the user intends to purchase.</param>
+    /// <param name="discount"> A <see cref="PromotionalOffer"/> to apply to the purchase.</param>
+    /// <param name="callback"> A <see cref="MakePurchaseCallback"/> completion block that is called when the purchase completes.</param>
+    ///
     public void PurchaseDiscountedPackage(Package package, PromotionalOffer discount, MakePurchaseFunc callback)
     {
         MakePurchaseCallback = callback;
         _wrapper.PurchasePackage(package, discount: discount);
     }
 
+    /// <summary>
+    /// Callback type for methods that return <see cref="CustomerInfo"/>.
+    /// Includes a <see cref="CustomerInfo"/> or an error.
+    /// </summary>
+    /// <param name="customerInfo"> A <see cref="CustomerInfo"/> if the request was successful, null otherwise. </param>
+    /// <param name="error"> An error if the request was not successful, null otherwise. </param>
+    public delegate void CustomerInfoFunc(CustomerInfo customerInfo, Error error);
     private CustomerInfoFunc RestorePurchasesCallback { get; set; }
 
+    /// <summary>
+    /// 
+    /// This method will post all purchases associated with the current Store account to RevenueCat and become
+    /// associated with the current <c>appUserID</c>. If the receipt is being used by an existing user, the current
+    /// <c>appUserID</c> will be aliased together with the <c>appUserID</c> of the existing user.
+    ///  Going forward, either <c>appUserID</c> will be able to reference the same user.
+    /// </summary>
```

Comment:
About this Should we add a link to the RC docs https://docs.revenuecat.com/docs/restoring-purchases ? Seems like it could be useful.

Reply:
Love it!! Will do this in the follow up PR too